src/policy/evaluator.ts

```diff
@@ -256,6 +257,34 @@ export class PolicyEvaluator {
 			}
 		}
 
+		if (conditions.only_has_authors_in) {
+			const { users, organizations, teams } = conditions.only_has_authors_in;
+			// const commits = await compareCommits(this.githubContext, trustedBranch, this.context.commits[0].sha);
+			const commits = this.context.commits;
+
+			// If there are no commits, return false
+			if (commits.length === 0) {
+				return false;
+			}
+
+			// Check that all commits were authored by the specified users/organizations/teams
+			const results = await Promise.all(
+				commits.map(async (commit: Commit) => {
+					return await this.isUserInAny(
+						commit.author?.login ?? '',
+						users ?? [],
+						organizations ?? [],
+						teams ?? [],
+					);
+				}),
+			);
+
+			this.logger.info(
+				`Evaluated condition only_has_authors_in: ${JSON.stringify(conditions.only_has_authors_in)}: ${results.every(Boolean)}`,
+			);
+			return results.every(Boolean);
```

Comment:
NIT: Can save a `.every` call by storing the result in a variable and using the variable in both places

Reply:
I made that change when I refactored this function in the following commit!